lib/identity_cache/configuration_dsl.rb

```diff
@@ -205,11 +205,20 @@ def build_recursive_association_cache(association, options) #:nodoc:
         options[:cached_accessor_name]   = "fetch_#{association}"
         options[:records_variable_name]  = "cached_#{association}"
 
-        self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
-          def #{options[:cached_accessor_name]}
-            fetch_recursively_cached_association('#{options[:records_variable_name]}', :#{association})
-          end
-        CODE
+        if method_defined?(options[:cached_accessor_name].to_sym)
+          self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
+            def #{options[:cached_accessor_name]}_with_idc
```

Comment:
Active record doesn't define fetch_polymorphic_records.  It looks like this is coming from SchemaChangeTest#test_embed_existing_cache_has_many which is making sure it is safe to change an association between being fully embedded or just embedding the id.  This warning is an indication of a problem in the tests, not the code under test.

Perhaps that can be done more cleanly using

```ruby
teardown_models
setup_models
```

which is done between tests.